modules/weko-records/weko_records/api.py

```diff
@@ -358,8 +369,42 @@ def update(cls, id_=0, name=None, schema=None, form=None, render=None,
                             'Invalid name: {}'.format(name))
                         raise ValueError(_('Invalid name.'))
                     item_type_name.name = name
-        return cls.create(item_type_name=item_type_name, name=name,
-                          schema=schema, form=form, render=render, tag=tag)
+                    updated_name = True
+                if upgrade_version or updated_name:
+                    return cls.create(item_type_name=item_type_name, name=name,
+                                      schema=schema, form=form, render=render,
+                                      tag=tag)
+                else:
+                    return cls.update(id_, schema, form, render)
+
+    @classmethod
+    def update(cls, item_type_id, schema, form, render):
+        """Update item type record.
+
+        @param item_type_id:
+        @param schema:
+        @param form:
+        @param render:
+        @return:
+        """
+        with db.session.begin_nested():
+            current_record = cls.get_by_id(id_=item_type_id)
+            record = cls(current_record.schema)
+            before_record_update.send(
+                current_app._get_current_object(),
+                record=record,
+            )
+            current_record.schema = schema
+            current_record.form = form
+            current_record.render = render
+            db.session.merge(current_record)
+            record.model = current_record
+        db.session.commit()
+        after_record_update.send(
+            current_app._get_current_object(),
+            record=record,
+        )
+        return record
```

Comment:
I think this function is as same as function commit. Can we reuse it?

Thanks.

Reply:
Yes, I have found a way to reuse commit function from update function, pls help to review it.